lib/benchee/benchmark/runner.ex

```diff
@@ -213,10 +209,14 @@ defmodule Benchee.Benchmark.Runner do
     do_benchmark(
       scenario,
       updated_context,
-      {[run_time | run_times], [memory_usage | memory_usages]}
+      {[run_time | run_times], updated_memory_usages(memory_usage, memory_usages) }
     )
   end
 
+  # We return nil if no memory measurement is performed so keep it empty
+  defp updated_memory_usages(nil, memory_usages), do: memory_usages
+  defp updated_memory_usages(memory_usage, memory_usages), do: [memory_usage | memory_usages]
```

Comment:
So this is where I moved the memory check. Idea was to avoid adding our "bad" memory values as early as possible so that bad values never actually end up in it (and we also avoid building a huge list of nils). Let me know what you think.

Reply:
Yeah, that's fine by me. At the time it didn't seem like they two would be likely to change together which is why I didn't extract anything, but I'm not opposed to this. It's small enough to easily undo if things do diverge later on.